### PR TITLE
INS-23479 update docs for Saas GI

### PR DESCRIPTION
**Pull request by hadarkorny**
INS-23202 - pull plugins from public repo 
INS-13762 - use client certificates for push plugins

### CODE REVIEW

**Review by ofer-haim — APPROVED** [review]

File: docs/UC_Configuration_GI.md [hunk 1]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
 
-**NOTE**: In GI, whenever using plug-ins that are based on Filebeat as a data shipper, the configured port should be 443. Guardium Insights will map this to an internal port
 
-4.  Copy the location of the downloaded certificate, in the ssl.certificate\_authorities row in step 3 of the Filebeat section on this page. For example, `ssl.certificate_authorities: ["/etc/pki/ca-trust/GuardiumInsightsCA.pem"]`
-    
-5.  Restart Filebeat to effect these changes
+1.  Open the Filebeat Configuration. (filebeat.yml). usually located in `/etc/filebeat/filebeat.yml`.
     
-    Linux: Enter the command: `sudo service filebeat restart`
+2.  Locate the `tags` section, enter the Data source tag. For example, `tags: ["any-mongodb"]`.
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:102]:
```suggestion
2.  Locate the `tags` section and enter the Data source tag. For example, `tags: ["any-mongodb"]`.
```

File: docs/UC_Configuration_GI.md [hunk 2]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
     
-6.  Windows: Restart in the Services window
+3.  Locate `output.logstash` section and add an entry for Guardium Insights:
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:104]:
```suggestion
3.  Locate the `output.logstash` section and add an entry for Guardium Insights:
```

File: docs/UC_Configuration_GI.md [hunk 3]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
+    ```
+    # The Logstash hosts
+    hosts: ["<hostname-URL>:443"]
+    ```
+**NOTE**: In GI, whenever using plug-ins that are based on Filebeat as a data shipper, the configured port should be 443. Guardium Insights will map this to an internal port
+
+4. Configure mTLS - Logstash to Filebeat:
+   1. Download the SSL certificate (`UC certificate authority`) from Guardium Insights and upload it to the machine where Filebeat is.
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:112]:
```suggestion
   a. Download the SSL certificate (`UC certificate authority`) from Guardium Insights and upload it to the machine where Filebeat is.
```

File: docs/UC_Configuration_GI.md [hunk 4]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
+   2. Copy the location of the downloaded certificate and enter it as the certificate authority.
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:113]:
```suggestion
   b. Copy the location of the downloaded certificate and enter it as the certificate authority.
```

File: docs/UC_Configuration_GI.md [hunk 5]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
+    ```
+    # List of root certificates for HTTPS server verifications
+    ssl.certificate_authorities: ["/etc/pki/ca-trust/GuardiumInsightsCA.pem"]
+    ```
+5. Configure mTLS - Filebeat to Logstash:
+   1. Create a filebeat certificate:
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:119]:
```suggestion
   a. Create a Filebeat certificate:
```

File: docs/UC_Configuration_GI.md [hunk 6]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
+   
+       a. Create a private key:
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:121]:
```suggestion
       i. Create a private key: 
```

File: docs/UC_Configuration_GI.md [hunk 7]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
+    ssl.certificate_authorities: ["/etc/pki/ca-trust/GuardiumInsightsCA.pem"]
+    ```
+5. Configure mTLS - Filebeat to Logstash:
+   1. Create a filebeat certificate:
+   
+       a. Create a private key: 
+       ```
+       openssl genrsa -out filebeat.key 2048 
+       ```
+       b. convert private key to pkc8
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:125]:
```suggestion
       ii. Convert private key to pkc8
```

File: docs/UC_Configuration_GI.md [hunk 8]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
+       ```
+       openssl pkcs8 -in filebeat.key -topk8 -out filebeat-pkcs8.key -nocrypt
+       ```
+       c. Creating certificate request file
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:129]:
```suggestion
       iii. Create a certificate request file
```

File: docs/UC_Configuration_GI.md [hunk 9]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
+       C = IL
+       CN = filebeat.lan
+    
+       [ req_ext ]
+       subjectAltName = @alt_names
+    
+       [ alt_names ]
+       DNS.1 = server.dns.com
+       ```
+       d. Now generate the certificate request file:
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:149]:
```suggestion
       iv. Now generate the certificate request file:
```

File: docs/UC_Configuration_GI.md [hunk 10]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
+       ```
+       openssl req -new -key filebeat.key -out filebeat.csr -config filebeat-csr.conf
+       ```
+       e. Generate certificate from csr file and CA
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:153]:
```suggestion
       v. Generate certificate from csr file and CA 
```

File: docs/UC_Configuration_GI.md [hunk 11]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
+       ```
+       authorityKeyIdentifier=keyid,issuer
+       basicConstraints=CA:FALSE
+       keyUsage = digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment
+       subjectAltName = @alt_names
+    
+       [alt_names]
+       DNS.1 = server.dns.com
+       ```
+       f. Now generate the certificate
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:164]:
```suggestion
       vi. Now generate the certificate
```

File: docs/UC_Configuration_GI.md [hunk 12]
```diff
@@ -87,19 +91,104 @@ The universal connector configuration has a few parts, all described in this tas
...
+       ```
+       openssl x509 -req -in filebeat.csr -CA filebeatCA.crt -CAkey filebeatCA.key -CAcreateserial -out filebeat.crt -days 365 -sha256 -extfile filebeat-cert.conf
+       ```
+   2. Locate `output.logstash` section and add an entry of the certificate:
```
Comment (AlexaNahum) [inline docs/UC_Configuration_GI.md:168]:
```suggestion
   b. Locate the `output.logstash` section and add an entry of the certificate:
```
(entry for the certificate?)

File: docs/UC_Configuration_GI.md (hunk 8 above)
Comment (hadarkorny) [reply docs/UC_Configuration_GI.md:130]: why?

**Pull request merged**